recipe/patches/0002-rely-on-DCMAKE_CUDA_ARCHITECTURES.patch

```diff
@@ -0,0 +1,439 @@
+-
+-    if("${QUICK_USER_ARCH}" STREQUAL "")
+-        
+-        # build for all supported CUDA versions
+-	if(${CUDAToolkit_VERSION} VERSION_EQUAL 8.0)
+-            message(STATUS "Configuring QUICK for SM3.0, SM5.0, and SM6.0")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM30FLAGS} ${SM50FLAGS} ${SM60FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
+-            
+-        elseif((${CUDAToolkit_VERSION} VERSION_GREATER_EQUAL 9.0) AND (${CUDAToolkit_VERSION} VERSION_LESS 10.0)) 
+-            message(STATUS "Configuring QUICK for SM3.0, SM5.0, SM6.0 and SM7.0")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM30FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
+-
+-        elseif((${CUDAToolkit_VERSION} VERSION_GREATER_EQUAL 10.0) AND (${CUDAToolkit_VERSION} VERSION_LESS 11.0))
+-            message(STATUS "Configuring QUICK for SM3.0, SM5.0, SM6.0, SM7.0 and SM7.5")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM30FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
+-
+-	elseif((${CUDAToolkit_VERSION} VERSION_EQUAL 11.0))
+-	    message(STATUS "Configuring QUICK for SM3.5, SM5.0, SM6.0, SM7.0, SM7.5 and SM8.0")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
+-
+-	elseif((${CUDAToolkit_VERSION} VERSION_GREATER_EQUAL 11.1) AND (${CUDAToolkit_VERSION} VERSION_LESS_EQUAL 11.7))
+-	    message(STATUS "Configuring QUICK for SM3.5, SM5.0, SM6.0, SM7.0, SM7.5, SM8.0 and SM8.6")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} ${SM86FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
+-	    
+-        elseif((${CUDAToolkit_VERSION} VERSION_GREATER_EQUAL 11.8) AND (${CUDAToolkit_VERSION} VERSION_LESS_EQUAL 12.0))
+-            message(STATUS "Configuring QUICK for SM5.0, SM6.0, SM7.0, SM7.5, SM8.0, SM8.6, SM9.0")
+-            list(APPEND CUDA_NVCC_FLAGS ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} ${SM86FLAGS} ${SM90FLAGS})
+-            list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
+-            set(DISABLE_OPTIMIZER_CONSTANTS TRUE)          
+-	else()
+-	    message(FATAL_ERROR "Error: Unsupported CUDA version. ${PROJECT_NAME} requires CUDA version >= 8.0 and <= 12.0.  Please upgrade your CUDA installation or disable building with CUDA.")
+-	endif()
++    list(APPEND CUDA_NVCC_FLAGS -DUSE_LEGACY_ATOMICS)
++    set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
```

Comment:
Indeed the code before was using `-DUSE_LEGACY_ATOMICS`, but that aspect should be unchanged by the patching (except to avoid the repetition per version).

Reply:
Except perhaps if the way to pass flags to nvcc has also changed... 😑

Reply:
Unfortunately yes. This was part of the CUDA language change. It is now called [`CMAKE_CUDA_FLAGS` to match the other `CMAKE_<LANG>_FLAGS`]( https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS.html#cmake-lang-flags )